src/main/java/com/minecolonies/coremod/colony/ColonyList.java

```diff
@@ -0,0 +1,271 @@
+
+    private int topID = 0;
+
+    private int size = 0;
+
+    /**
+     * Get the Colony with the provided colony id.
+     * @param index colony id.
+     * @return The Colony associated with the provided id.
+     */
+    @Nullable
+    // no way to remove this, java does it too
+    @SuppressWarnings("unchecked")
+    public T get(int index)
+    {
+        if (index < 1 || index >= list.length)
+        {
+            return null;
+        }
+
+        return (T) list[index];
+    }
+
+    /**
+     * Creates a new Colony, adds it to the list, and returns it.
+     * @param world The world for the Colony.
+     * @param position The position for the Colony center.
+     * @return The newly created Colony.
+     */
+    @NotNull
+    public Colony create(World world, BlockPos position)
+    {
+        final int colonyID = getNextColonyID();
+        if (colonyID >= list.length)
+        {
+            expandList();
+        }
+
+        if(list[colonyID] != null)
+        {
+            throw new IllegalArgumentException(
+                    String.format("Already a colony registered to id=%d, colony=%s, not creating new colony",
+                            colonyID,
+                            list[colonyID].getName()));
+        }
+
+        final Colony colony = new Colony(colonyID, world, position);
+        size++;
+        list[colony.getID()] = colony;
+        return colony;
+    }
+
+    /**
+     * Add a new Colony to the List.
+     * @param colony colony to add to the list.
+     */
+    public void add(T colony)
+    {
+        final T existingColony = get(colony.getID());
+        if (existingColony != null && existingColony != colony)
+        {
+            throw new IllegalArgumentException(
+                    String.format("Already a colony registered to id=%d, colony=%s, not changing to colony=%s",
+                            colony.getID(),
+                            existingColony.getName(),
+                            colony.getName()));
+        }
+
+        while (colony.getID() >= list.length)
+        {
+            expandList();
+        }
+
+        int emptyIds = colony.getID() - 1;
+        while(emptyIds > 0  && list[emptyIds] == null)
+        {
+            nullIndices.add(emptyIds);
+            emptyIds--;
+        }
+
+        size++;
+        topID = colony.getID();
+
+        list[colony.getID()] = colony;
+    }
+
+    /**
+     * Remove the Colony from the list.
+     * @param colony the Colony to remove.
+     */
+    public void remove(T colony)
+    {
+        remove(colony.getID());
+    }
+
+    /**
+     * Remove the colony with the provided id from the list.
+     * @param id colony id to remove.
+     */
+    public void remove(int id)
+    {
+        if (list[id] == null)
+        {
+            Log.getLogger().warn("Tried to remove colony with id=%d, but it didn't exist.", id);
+        }
+
+        size--;
+        list[id] = null;
+
+        if (!nullIndices.contains(id))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)